pydatastructs/trees/binary_trees.py

```diff
@@ -415,6 +415,107 @@ def rank(self, x):
             walk = p
         return r
 
+    def _simple_path(self, key, root, path):
+        """
+        Utility funtion to find the simple path between root and node.
+        """
+        if root is None:
+            return False
+        path.append(root)
+        if self.tree[root].key == key:
+            return True
+
+        if self._simple_path(key, self.tree[root].left, path) or \
+            self._simple_path(key, self.tree[root].right, path):
+            path.append(root)
+            return True
+
+        path.pop()
+        return False
+
+    def simple_path(self, key):
+        path = []
```

Comment:
Looks like `path` is a stack. Can we use, `pydatastructs.Stack` here?

Reply:
Nope. After returning ill be iterating over the list to find first not matched item. But we cant iterate in stack.